advanced/event.yaml

```diff
@@ -0,0 +1,90 @@
+components:
+  schemas:
+    Event.Request:
+      type: object
+      properties:
+        type:
+          description: |-
+            Constrain the events being returned by specifying event types.
+            If the list is empty then all event types will be returned. 
+          type: array
+          items: 
+            type: string
+            enum:
+            - link_down
+            - link_up
+            - route_withdraw
+            - route_advertise
+            - flow_rx_rate_above_threshold
+            - flow_rx_rate_below_threshold
+            example: route_withdraw
+        source:
+          description: |-
+            Constrain the events being returned by specifying event sources. 
+            If the list is empty then all event sources will be returned. 
+          type: array
+          items:
+            type: string
+            x-constraint:
+            - '#/components/schemas/Port/properties/name'
+            - '#/components/schemas/Device.Bgpv4Route/name'
+            - '#/components/schemas/Device.Bgpv6Route/name'
+
+    Event: 
+      description: |-
+        A container for an event that has occurred in the system.
+      type: object
+      properties:
+        type:
+          description: |-
+            The type of event that occurred.
+          type: string
+          enum:
+          - link_down
+          - link_up
+          - route_withdraw
+          - route_advertise
+          - flow_rx_rate_above_threshold
+          - flow_rx_rate_below_threshold
+          example: route_withdraw
+        source:
+          description: |-
+            The source of the event. 
+            The source MUST be the value of one of the x-constraint paths,
+            which means the source MUST be a unique name in the configuration. 
+          type: string
+          x-constraint:
+          - '#/components/schemas/Port/properties/name'
+          - '#/components/schemas/Flow/properties/name'
+          - '#/components/schemas/Device.Bgpv4Route/properties/name'
+          - '#/components/schemas/Device.Bgpv6Route/properties/name'
+          example: Bgpv4 Route Range 1
+        begin_timestamp_ns:
+          description: |-
+            The time in seconds at which the event began
+          type: number
+          format: double
+        end_timestamp_ns:
+          description: |-
+            The time in seconds at which the event ended
```

Comment:
Is this in seconds or nanoseconds?

Reply:
Fixing description to reflect nanoseconds.